### PR TITLE
KNN Codec versioning support for backward compatibility

### PR DESCRIPTION
**Pull request by vamshin**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/k-NN/issues/20

*Description of changes:*
KNN Codec versioning support for backward compatibility

*Testing*
Integ tests passed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by zhanghg08 — APPROVED** [review]
LGTM, thanks!

File: src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java [hunk 1]
```diff
@@ -126,6 +128,7 @@ public void testCorrectStatsReturned() throws IOException {
      * @throws IOException throws IOException
      */
     public void testStatsValueCheck() throws IOException {
+// <<<<<<< HEAD
```
Comment (jmazanec15) [inline src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: Looks like leftover merge conflict

File: src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java [hunk 2]
```diff
@@ -232,6 +235,118 @@ public void testStatsValueCheck() throws IOException {
         assertTrue("Miss and hit count does not return expected",
                 (Integer) metricMap2.get(StatNames.HIT_COUNT.getName()) == initialHitCount + 1 &&
                         (Integer) metricMap2.get(StatNames.MISS_COUNT.getName()) == initialMissCount + 1);
+//=======
```
Comment (jmazanec15) [inline src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: Why is this section commented out?

Comment (vamshin) [reply src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: Fixed

File: src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java (hunk 1 above)
Comment (vamshin) [reply src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: fixed

**Review by jmazanec15 — APPROVED** [review]
LGTM thanks!

**Pull request merged**